src/verify.rs

```diff
@@ -49,37 +46,59 @@ pub async fn dev_portal_verify_proof<V: alloy::sol_types::SolValue + Send>(
 	signal: V,
 	developer_portal_url: String,
 ) -> Result<(), Error> {
-	let signal = signal.abi_encode_packed();
+	let packed = signal.abi_encode_packed();
+	let signal_hash = if packed.is_empty() {
+		None
+	} else {
+		Some(format!("0x{:x}", hash_to_field(&packed)))
+	};
+
+	let body = VerificationRequest {
+		action: action.to_owned(),
+		proof: proof.proof.clone(),
+		merkle_root: proof.merkle_root.clone(),
+		nullifier_hash: proof.nullifier_hash.clone(),
+		verification_level: proof.verification_level,
+		signal_hash,
+	};
+
+	let client = reqwest::Client::new();
+	let url = format!("{developer_portal_url}/api/v2/verify/{app_id}");
 
-	let response = reqwest::Client::new()
-		.post(format!("{developer_portal_url}/api/v2/verify/{app_id}"))
-		.header(header::USER_AGENT, "idkit-rs")
-		.json(&VerificationRequest {
-			proof: proof.proof,
-			signal_hash: if signal.is_empty() {
-				None
-			} else {
-				Some(format!("0x{:x}", hash_to_field(&signal)))
+	let policy = ExponentialBuilder::default()
+		.with_min_delay(Duration::from_millis(100))
+		.with_max_delay(Duration::from_secs(2))
+		.with_jitter()
+		.with_max_times(3);
+
+	let attempt = || async {
+		let resp = client
+			.post(&url)
+			.header(header::USER_AGENT, "idkit-rs")
+			.json(&body)
+			.send()
+			.await?;
```

Comment:
```suggestion
			.await.map_err(Error::Http)?;
```

It's better to unwrap into a specific error for easier debugging in the future. Otherwise it just returns a generic error